packages/drivers/odsp-driver/src/vroom.ts

```diff
@@ -40,17 +40,21 @@ interface IJoinSessionBody {
 export async function fetchJoinSession(
 	urlParts: IOdspUrlParts,
 	path: string,
-	method: string,
+	method: "GET" | "POST",
 	logger: ITelemetryLoggerExt,
-	getStorageToken: InstrumentedStorageTokenFetcher,
+	getAuthHeader: InstrumentedStorageTokenFetcher,
 	epochTracker: EpochTracker,
 	requestSocketToken: boolean,
 	options: TokenFetchOptionsEx,
 	disableJoinSessionRefresh: boolean | undefined,
 	isRefreshingJoinSession: boolean,
 	guestDisplayName?: string,
 ): Promise<ISocketStorageDiscovery> {
-	const token = await getStorageToken(options, "JoinSession");
+	const siteOrigin = getOrigin(urlParts.siteUrl);
+	const url = `${getApiRoot(siteOrigin)}/drives/${urlParts.driveId}/items/${
+		urlParts.itemId
+	}/${path}?ump=1`;
+	const authHeader = await getAuthHeader({ ...options, request: { url, method } }, "JoinSession");
```

Comment:
Could method be 'GET' here? How does it reconcile with using multipart/form when making request?

Reply:
I do not know. As far as I can see it's always called with POST but the intent is clearly that method is determined by the caller. Regardless, in this case, all we need just to narrow the types from string.